pkg/prometheus/statefulset.go

```diff
@@ -600,19 +600,43 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 	var readinessProbeHandler v1.Handler
 	var livenessFailureThreshold int32
 	if (version.Major == 1 && version.Minor >= 8) || version.Major == 2 {
-		livenessProbeHandler = v1.Handler{
-			HTTPGet: &v1.HTTPGetAction{
-				Path: path.Clean(webRoutePrefix + "/-/healthy"),
-				Port: intstr.FromString(p.Spec.PortName),
-			},
+		{
+			healthyPath := path.Clean(webRoutePrefix + "/-/healthy")
+			if p.Spec.ListenLocal {
+				livenessProbeHandler.Exec = &v1.ExecAction{
+					Command: []string{
+						"wget",
```

Comment:
nit: how can we make sure this is available in all images?

Reply:
It is a posix thing. I'm afraid we can't really make sure. However, even the busybox based Prometheus image had wget and not curl.
Unfournately, I just saw that it's exactly the other way around for RHEL based images... 

:roll_eyes:

Reply:
Discussing offline, we need to somehow check for either of wget or curl being available and then do the probe. We should fail if none available.

Reply:
Not sure if we should fail, thats not backwards compatible. Also failing implies the liveness probe failed. hmm I guess curl and wget should cover 90% of usecases? For the rest can't we add a way to disable the adding of liveness probe?

Reply:
I'd argue that we would want to fail hard in this case, rather than silently not probe at all.

Reply:
crazy idea: can we implement a "wget"/"curl" request functionality in the prometheus operator binary itself as a separate subcommand? Shipping the operator binary is the lowest common denominator anyways.

Reply:
Great idea :flushed: :smile: :blush: 
Shouldn't be too hard indeed.

I'll give it a shot.

Reply:
Wait. Actually that doesn't work. We need something inside the Prometheus Container, not the Operator's Container.

Reply:
ohhh, ok, right. maybe we can persuade upstream then? :) (not sure about this)

Reply:
I think we are fine with checking for wget first, if not there, using curl, and failing. Is there any image we know of that doesn't have either one of those?